src/Microsoft.AspNetCore.Mvc.Core/ModelBinding/Binders/CollectionModelBinderProvider.cs

```diff
@@ -43,8 +43,6 @@ public IModelBinder GetBinder(ModelBinderProviderContext context)
             // If the model type is IEnumerable<> then we need to know if we can assign a List<> to it, since
             // that's what we would create. (The cases handled here are IEnumerable<>, IReadOnlyColection<> and
             // IReadOnlyList<>).
-            //
-            // We need to check IsReadOnly because we need to know if we can SET the property.
             var enumerableType = ClosedGenericMatcher.ExtractGenericInterface(modelType, typeof(IEnumerable<>));
             if (enumerableType != null && !context.Metadata.IsReadOnly)
```

Comment:
Do not include this check. Both of those properties will not work (defined in this case as a binding that leaves `Unvalidated` entries in the `ModelStateDictionary` anyhow -- because the `ComplexTypeModelBinder` will be used but ignore them.